asserts/ifacedecls.go

```diff
@@ -397,6 +404,31 @@ func baseCompileConstraints(context *subruleContext, cDef constraintsDef, target
 		}
 		target.setOnClassicConstraint(c)
 	}
+	onCoreDesktop := cMap["on-core-desktop"]
+	if onCoreDesktop == nil {
+		defaultUsed++
+	} else {
+		var c *OnCoreDesktopConstraint
+		switch x := onCoreDesktop.(type) {
+		case string:
+			switch x {
+			case "true":
+				c = &OnCoreDesktopConstraint{CoreDesktop: true}
+			case "false":
+				c = &OnCoreDesktopConstraint{CoreDesktop: false}
+			}
+		case []interface{}:
+			lst, err := checkStringListInMap(cMap, "on-core-desktop", fmt.Sprintf("on-core-desktop in %s", context), validDistro)
+			if err != nil {
+				return err
+			}
+			c = &OnCoreDesktopConstraint{CoreDesktop: true, SystemIDs: lst}
```

Comment:
Does this case make sense? The only way `on-core-desktop` can be true is if the distro is Ubuntu Core. So the simple boolean version is probably all that matters.

Reply:
I wasn't sure - I figure there could be a family of core desktops so this might be useful? I guess since this could be added at a later date it can be dropped for now.